### ISSUE TITLE
Cached Virtual Files Should Not Be Optimistically Deleted 

### ISSUE THREAD

**Issue by akshayka** [marimo-team/marimo#314]
### Describe the bug

Virtual Files are being optimistically deleted on cell re-run. This breaks if a user is caching HTML that used a virtual file created in a previous run of the cell.

This bug is very similar to the bug raised in #307 .

In the below video, notice how the plot doesn't update when the slider goes to the value $2$: this is because the virtual file for the quadratic plot was incorrectly deleted.

[cached_virtual_files_bug.webm](https://github.com/marimo-team/marimo/assets/1994308/3840ee20-9cc1-4526-a846-e2f2d6cda2ff)

### Version

0.1.45

### Code to reproduce

```python
import marimo

__generated_with = "0.1.43"
app = marimo.App()


@app.cell
def __(exp):
    exp
    return


@app.cell
def __(altp, exp, mo, mplplot, plotlyplot, snsplot):
    mo.vstack(
        [
            mo.md(f"$$y=x^{exp.value}$$"),
            mo.tabs(
                {
                    "altair": mo.vstack([altp, altp.value]),
                    "plotly": plotlyplot(exp.value),
                    "seaborn": snsplot(exp.value),
                    "matplotlib": mplplot(exp.value),
                }
            ),
        ]
    )
    return


@app.cell
def __():
    import marimo as mo
    from plotly import express as px
    import altair as alt
    import numpy as np
    import pandas as pd
    import seaborn as sns
    from matplotlib import pyplot as plt
    from functools import cache
    return alt, cache, mo, np, pd, plt, px, sns


@app.cell
def __(mo):
    exp = mo.ui.slider(1, 8, 1, label="exp: ",value=8)
    return exp,


@app.cell
def __(cache, np):
    @cache
    def prep_xy(exp):
        x = np.linspace(0, 5, 100)
        return x, x**exp
    return prep_xy,


@app.cell
def __(alt, cache, mo, pd, prep_xy):
    @cache
    def altplot(exp):
        x, y = prep_xy(exp)
        data = pd.DataFrame({"x": x, "y": y})
        return mo.ui.altair_chart(
            alt.Chart(data).mark_line().encode(x="x", y="y"),
            chart_selection="interval",
        )
    return altplot,


@app.cell
def __(cache, prep_xy, px):
    @cache
    def plotlyplot(exp):
        x, y = prep_xy(exp)
        return px.line(x=x, y=y)
    return plotlyplot,


@app.cell
def __(cache, prep_xy, sns):
    @cache
    def snsplot(exp):
        x, y = prep_xy(exp)
        return sns.lineplot(x=x, y=y)
    return snsplot,


@app.cell
def __(cache, mo, plt, prep_xy):
    @cache
    def mplplot(exp):
        x, y = prep_xy(exp)
        plt.plot(x, y)
        return mo.mpl.interactive(plt.gcf())
    return mplplot,


@app.cell
def __(altplot, exp):
    altp = altplot(exp.value)
    return altp,


if __name__ == "__main__":
    app.run()
```

**Comment by akshayka** [comment]
@proltheus FYI this is another bug that your repro unearthed. We're working on a fix.

**Comment by akshayka** [comment]
Fixed by #320 , released in 0.1.47.
